scrapy/link.py

```diff
@@ -28,6 +28,7 @@ def __init__(self, url, text='', fragment='', nofollow=False):
         self.text = text
         self.fragment = fragment
         self.nofollow = nofollow
+        self.encoding = encoding
 
     def __eq__(self, other):
         return self.url == other.url and self.text == other.text and \
```

Comment:
hm, adding encoding to comparison makes a lot of tests fail because many HtmlResponse in tests get detected as cp1252 without explicit encoding (while I was expecting UTF-8 by default)